crates/subspace-networking/examples/benchmark.rs

```diff
@@ -159,6 +159,7 @@ pub async fn configure_dsn(
         kademlia_mode: Some(Mode::Client),
         request_response_protocols: vec![PieceByIndexRequestHandler::create(|_, _| async { None })],
         bootstrap_addresses,
+        enable_autonat: false,
```

Comment:
Why would we want to disable it, for what use case? Less non-optional stuff would be easier to maintain.

Reply:
This line is a good answer to your question. We don't want that temporary clients (like benchmark or other DSN tests) to spoil autonat probes and thus provide incorrect data to peers. Another potential candidate would be `subspace-bootstrap-node` to reduce the connection queue of the peer, however, I don't think it's an issue right now.

Reply:
Doesn't make a lot of sense to me. Even temporary clients are perfectly capable of checking whether particular instance has reachable public address :man_shrugging:

Reply:
Some DSN clients are ephemeral: they could accept the inbound probe and die after that - leaving the probe with an error message contributing to the  "private address" decision.

Reply:
Then the issue is that we treat error from disconnected peer as "private address", we clearly shouldn't do that

Reply:
Autonat doesn't flip the status immediately. It waits for the "confidence" to decrease to zero first:  https://github.com/libp2p/rust-libp2p/blob/16b2f1563d2d16c4e35cc8c8024dca7764a88598/protocols/autonat/src/behaviour/as_client.rs#L343

Reply:
On the second look, it seems to ignore failed probes altogether: https://github.com/libp2p/rust-libp2p/blob/16b2f1563d2d16c4e35cc8c8024dca7764a88598/protocols/autonat/src/behaviour/as_client.rs#L365

Treating only reported failed dials as meaningful information. 

I would still make the protocol optional. Why make ephemeral clients have more features than they should?

Reply:
I just don't like to make many things optional, too many invariants to deal with